cmake/frida.cmake

```diff
@@ -26,6 +26,11 @@ elseif(${FRIDA_OS_ARCH} MATCHES "linux-arm.*")
   set(FRIDA_GUM_DEVKIT_SHA256 "6b5963eb740062aec6c22c46ec2944a68006f72d290f714fb747ffe75b448a60")
   # Frida only has armhf builds..
   set(FRIDA_OS_ARCH "linux-armhf")
+elseif(${FRIDA_OS_ARCH} MATCHES "darwin-arm.*")
```

Comment:
Are you getting `FRIDA_OS_ARCH` `darwin-arm64e`?

From my understanding, it is gonna be `darwin-arm64e` because this string is built from the `CMAKE_SYSTEM_PROCESSOR` variable, which is x86 by default, but it can detect arm by setting the env var [CMAKE_APPLE_SILICON_PROCESSOR](https://cmake.org/cmake/help/v3.23/variable/CMAKE_APPLE_SILICON_PROCESSOR.html), but then `CMAKE_SYSTEM_PROCESSOR` is `arm64`, not `arm64e`.

On my end, to get it compiling on Mac, I only had to set `CMAKE_APPLE_SILICON_PROCESSOR` and added the branch
```cmake
elseif(${FRIDA_OS_ARCH} MATCHES "darwin-arm64")
  set(FRIDA_CORE_DEVKIT_SHA256 "7811e516e6b7bbc0153d30095560e0b1133f154060c5542764100d3e0eb2ab2b")
  set(FRIDA_GUM_DEVKIT_SHA256 "03f6085ae5330cf38e0a498784500675fc5bd7361bb551a9097ba5fe397aceda")
else()
```
Given it doesn't seem `CMAKE_SYSTEM_PROCESSOR` ever gets the `arm64e` value

Reply:
Hi @viniciusd , arm64e was on the release link of frida-gum, I had explored both, arm64 and arm64e, and here is the difference between both: https://stackoverflow.com/a/65918986/7116645

Reply:
Yes, it makes sense to be using the arm64e, but my point is more on the `FRIDA_OS_ARCH` value, is it ever `darwin-arm64e`? From what I could gather, it will never come with the extra `e` because the string is built from `CMAKE_SYSTEM_PROCESSOR`